### PR TITLE
Close DRBD handshake race in AssembleInstanceDisks

### PR DESCRIPTION
**Pull request by rbott**
This PR addresses #846 in a way that does not introduce new cluster parameters. 

The two-pass assemble only narrowed the window where a DRBD device could be promoted while still in WFConnection. Between passes, reuse the existing `drbd_wait_sync` RPC on the secondaries to wait for the handshake; failures are logged but do not abort. The logic does of course not apply to non-DRBD blockdevices.

To avoid new cluster parameters (for now), we resort to document `net-custom` as the supported way to override the hardcoded after-sb-*pri policies. With DRBD utils, the latest occurence of CLI parameters always "wins", hence it is possible to override the cluster default policies.

### CODE REVIEW

File: lib/cmdlib/instance_storage.py [hunk 1]
```diff
@@ -1623,7 +1623,26 @@ def AssembleInstanceDisks(lu, instance, disks=None, ignore_secondaries=False,
...
+  # instances without secondaries (single-node / diskless).
+  if any(d.dev_type == constants.DT_DRBD8 for d in disks):
+    secondary_nodes = lu.cfg.GetInstanceSecondaryNodes(instance.uuid)
+    if secondary_nodes:
+      wait_result = lu.rpc.call_drbd_wait_sync(secondary_nodes,
+                                               (disks, instance))
+      for node_uuid, nres in wait_result.items():
+        node_name = lu.cfg.GetNodeName(node_uuid)
+        if nres.fail_msg:
+          lu.LogWarning("DRBD handshake wait failed on secondary %s: %s;"
```
Comment (apoikos) [inline lib/cmdlib/instance_storage.py]: If we are to provide strong guarantees against data loss, this should be a hard failure rather than a simple warning,  or at least fail hard unless `ignore_secondaries` is `True`. If we were to turn this to a hard failure, what potential breakage would we expect?

**Review by apoikos — COMMENTED** [review]
Thanks for looking into this @rbott!

**Comment by rbott** [comment]
Thanks, trying to clear all items on the 3.2 roadmap/milestone list. Although I allowed myself to move the q35/UEFI stuff to 3.3 (as that is probably to big and will take some more time to implement properly).

File: lib/cmdlib/instance_storage.py (hunk 1 above)
Comment (rbott) [reply lib/cmdlib/instance_storage.py]: You're absolutely right, we should hard-fail here. I have added some code to distinguish the "non-error" cases and tried to add descriptive error/warning messages for each scenario. 

The QA suite is currently running, but that will take several hours to complete

Comment (rbott) [reply lib/cmdlib/instance_storage.py]: (but it probably needs more manual testing anyways to cover corner cases)

Comment (rbott) [reply lib/cmdlib/instance_storage.py]: The QA suite passed on Debian Trixie and Forky with KVM / DRBD backends. In general I'd say there is no breakage to expect with regular operations. Also the special cases (secondary node offline etc.) should be covered by the updated code and not break emergency operations.

**Review by apoikos — APPROVED** [review]
LGTM, thanks!
